File: CSharp/Eta/Applications/PerfTools/Common/ItemRequest.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license      --
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.  --
 *|                See the project's LICENSE.md for details.                  --
 *|           Copyright (C) 2022 Refinitiv. All rights reserved.              --
 *|-----------------------------------------------------------------------------
 */

using Refinitiv.Eta.Codec;

namespace Refinitiv.Eta.PerfTools.Common
{
    /// <summary>
    /// Item request information
    /// </summary>
    public class ItemRequest
    {
        /// <summary>
        /// Item state
        /// </summary>
        public ItemRequestState RequestState { get; set; }
        /// <summary>
        /// Storage for the item's name
        /// </summary>
        public string? ItemName { get; set; }
        /// <summary>
        /// Structure containing item information
        /// </summary>
        public ItemInfo ItemInfo { get; set; }
        /// <summary>
        ///Message key used to request this item
        /// </summary>
        public IMsgKey MsgKey { get; set; }

        public ItemRequest()
        {
            MsgKey = new MsgKey();
            ItemInfo = new ItemInfo();
        }
        public void Clear()
        {
            MsgKey.Clear();
            ItemInfo.Clear();
            RequestState = ItemRequestState.NOT_REQUESTED;
            ItemInfo.Attributes.MsgKey = MsgKey;
        }
    }
}

File: Java/Eta/Applications/PerfTools/src/main/java/com/refinitiv/eta/perftools/consperf/ItemRequest.java
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2019-2022 LSEG. All rights reserved.     
 *|-----------------------------------------------------------------------------
 */

package com.refinitiv.eta.perftools.consperf;

import com.refinitiv.eta.codec.CodecFactory;
import com.refinitiv.eta.codec.MsgKey;
import com.refinitiv.eta.perftools.common.ItemInfo;

/** Item request information. */
public class ItemRequest
{
	private int _requestState;	/* Item state. */
	private String _itemName;	/* Storage for the item's name. */
	private ItemInfo _itemInfo;	/* Structure containing item information. */
	private MsgKey _msgKey;		/* Message key used to request this item. */
	
	{
		_msgKey = CodecFactory.createMsgKey();
		_itemInfo = new ItemInfo();
	}
	
	/** Clears an ItemRequest. */
	public void clear()
	{
		_msgKey.clear();
		_itemInfo.clear();
		_requestState = ItemRequestState.NOT_REQUESTED;
		_itemInfo.attributes().msgKey(_msgKey);
	}

	/**
	 *  Item state.
	 *
	 * @return the int
	 */
	public int requestState()
	{
		return _requestState;
	}

	/**
	 *  Item state.
	 *
	 * @param requestState the request state
	 */
	public void requestState(int requestState)
	{
		_requestState = requestState;
	}

	/**
	 *  Storage for the item's name.
	 *
	 * @return the string
	 */
	public String itemName()
	{
		return _itemName;
	}

	/**
	 *  Storage for the item's name.
	 *
	 * @param itemName the item name
	 */
	public void itemName(String itemName)
	{
		_itemName = itemName;
	}

	/**
	 *  Structure containing item information.
	 *
	 * @return the item info
	 */
	public ItemInfo itemInfo()
	{
		return _itemInfo;
	}

	/**
	 *  Structure containing item information.
	 *
	 * @param itemInfo the item info
	 */
	public void itemInfo(ItemInfo itemInfo)
	{
		_itemInfo = itemInfo;
	}

	/**
	 *  Message key used to request this item.
	 *
	 * @return the msg key
	 */
	public MsgKey msgKey()
	{
		return _msgKey;
	}

	/**
	 *  Message key used to request this item.
	 *
	 * @param msgKey the msg key
	 */
	public void msgKey(MsgKey msgKey)
	{
		_msgKey = msgKey;
	}
}
